packages/pyright-internal/src/analyzer/parseTreeUtils.ts

```diff
@@ -96,25 +96,46 @@ export function findNodeByOffset(node: ParseNode, offset: number): ParseNode | u
 
     // The range is found within this node. See if we can localize it
     // further by checking its children.
-    const children = getChildNodes(node);
+    let children = getChildNodes(node);
+    if (isCompliantWithNodeRangeRules(node) && children.length > 20) {
```

Comment:
I asked chatGPT generally when binary search would be faster than linear search and answer was generally 20 elements, so choose. 20 :)